### ISSUE TITLE
Enhancement: Persistent connection support

### ISSUE THREAD

**Issue by erikdemarco** [pressjitsu/pj-page-cache-red#35]
Please add a support for persistent connection, so instead using:
$redis->connect
we use 
$redis->pconnect

Using persistent connection will not reinitialize redis driver on each request
So It will be faster.

**Comment by soulseekah** [comment]
Persistent connections are experimental and can be set only in threaded version of PHP (i.e. it won't probably work in FPM (subprocess) mode).

Also starting from version 4.2.1 of the PHP Redis libarary, it became possible to use connection pooling by setting INI variable redis.pconnect.pooling_enabled to 1.

We're not going to be support explicit pconnect until it's stable enough and works with FPM environments.

If you have any other suggestions, feel free to chip in.

**Comment by erikdemarco** [comment]
@soulseekah 

> (i.e. it won't probably work in FPM (subprocess) mode)

Lets try a real world example. I tested it using W3 Total Cache with php-fpm default settings. W3TC redis page cache can use both mode persistent and non persistent (default to non persistent).

The result for page containing 100 products list with 100visits/ secs for 1 minute:
Without cache: ~3500ms
With persistent connection: ~600ms
Without persistent connection: ~700ms

So it works faster with persistent connection even in fpm.

> "We're not going to be support explicit pconnect until it's stable enough"

Lets look at real world example again with most popular plugins:
-) W3 Total cache defaulted to use persistent connection for redis (The most popular page cache which can use redis)
-) Redis Object Cache By Till Krüss only support persistent connection 

Both of them already been battle tested. I looked at their wp support forum and there is nothing mentioning problem with persistent connection

`If you have any other suggestions, feel free to chip in.`

My suggestion is to make it modifiable to use either mode via pj_user_config.
